### PR TITLE
OCM-2369 | feat : Check PR Github Action

### PR DESCRIPTION
**Pull request by hunterkepley**
JIRA: [OCM-2369](https://issues.redhat.com/browse/OCM-2369)

This jira is about adding common github actions/CI to this repo to help with PRs, commits, etc ensuring that they are labelled properly and look good. This will help out future development greatly.

### CODE REVIEW

File: .github/workflows/check-pull-request.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+          - macos-latest
+          - windows-latest
+    runs-on: ${{ matrix.platform }}
+    steps:
+      - name: Checkout the source
+        uses: actions/checkout@v3
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.18
```
Comment (ciaranRoche) [inline .github/workflows/check-pull-request.yml:45]: go-mod is 1.19 should we update these to match?

Comment (hunterkepley) [reply .github/workflows/check-pull-request.yml:45]: ah yes, let me do that

File: .github/workflows/check-pull-request.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
+#
+# Copyright (c) 2021 Red Hat, Inc.
```
Comment (gdbranco) [inline .github/workflows/check-pull-request.yml:2]:
```suggestion
# Copyright (c) 2023 Red Hat, Inc.
```

File: .github/workflows/check-pull-request.yml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - name: Ensure dependencies are downloaded
+        run: go mod download
+      - name: Setup Ginkgo
+        run: go install github.com/onsi/ginkgo/v2/ginkgo@v2.9
```
Comment (ciaranRoche) [inline .github/workflows/check-pull-request.yml:58]: Update ginkgo to the same version as in go mod `v2.11.0`

**Comment by ciaranRoche** [comment]
lgtm

**Pull request merged**